import Signals

def init():
    # Initialize input signals
    Signals.CAN_Signals.update({
        "targetContactorsStatus": "open",
        "evConnectionState": "disconnected",
        "chargingState": "standby",
        "targetVoltage": 0,
        "targetCurrent": 0,
        "batteryStateOfCharge": 0,
        "cableCheckVoltage ": 0,
        "vehicleId": 0,
        "chargeMode": "unknown",
        "evMinPower": 0,
        "evMinCurrent": 0,
        "evMinVoltage": 0,
        "evMaxCurent": 0,
        "evMaxPower": 0,
        "evMaxVoltage": 0,
        "chargingProfileMaxPowerLimit": 0,
        "timeToFullSoc": 0,
        "CHRGR_iAcAct": "Init",
        "CHRGR_iAcAct2": "Init",
        "CHRGR_iAcAct3": "Init",
        "CHRGR_uAcActPha1": "Init",
        "CHRGR_uAcActPha2": "Init",
        "CHRGR_uAcActPha3": "Init",
        "CHRGR_iDcAct": 0,
        "CHRGR_stOperAct": 0,
        "CHRGR_uDcAct": 0,
        "CHRGR_stErr": "no warning",
        "CHRGR_stHvIntlck": "open",
        "DC1_bReqMIL": "MIL Off",
        "DC1_stT30c": "init",
        "DC1_Temp": 0,
        })
    
    Signals.HMI_Signals.update({
        "HealthIcon": "No icon",
        })
    
    # Initialize output signals
    Signals.CAN_Signals.update({
        "mc_VCU_OBCControl": 0,
        "VCU_OBCMode_Cmd": "Init",
        "VCU_uDcReqChrgr": "Init",
        "VCU_iDcLimChrgr": "Init",
        "VCU_iAcMaxAllwdChrgr": "Init",
        "CRC_VCU_OBCControl": 0,
        "CrashDetect": "No Crash",
        "CrashConfirmed_Sts": "Not Active",
        "mc_ACU_Crash": 0,
        "CRC_VCU_OBCControl": 0,
        "NWM_WkUp_BCU_Rx": "Default",
        "NWM_WkUp_BCU_Reason": "Network",
        "NWM_WkUp_BCU_NodeAddress": 0,
        "NWM_Sleep_BCU_Req": "Default",
        "mc_VCU_DCDCControl": 0,
        "VCU_VoltageDemandHV": "Init",
        "VCU_CurrentLimitHV": "Init",
        "VCU_CurrentLimitLV": "Init",
        "VCU_VoltageDemandLV": "Init",
        "VCU_DCDCMode_Cmd": "Standby",
        "VCU_stHvDeactvnDC": "no deactivation",
        "CRC_VCU_DCDCControl": 0,
        "mc_VCU_SysControl": 0,
        "KL15_VCU_Sts": "ON",
        "HV_HVIL_Sts": "HVIL Open",
        "CRC_VCU_SysControl": 0,
        "limitVoltageMin": 0,
        "limitVoltageMax": 600,
        "limitPowerMax": 11000,
        "limitPowerMin": 0,
        "limitCurrentMin": 0,
        "limitCurrentMax": 25,
        "contactorsStatus": "open",
        "operationalStatus": "operative",
        "isolationStatus": "invalid",
        "drivenVoltage": 0,
        "drivenCurrent": 0,
        "temperature": 0,
        "measuredVoltage": 0,
        "measuredCurrent": 0,
        })
    
    Signals.MaxOutPwr = 7000